Lib/test/test_concurrent_futures.py

```diff
@@ -456,6 +489,24 @@ def test_max_workers_negative(self):
                                         "than 0"):
                 self.executor_type(max_workers=number)
 
+    def test_free_reference(self):
+        # Issue #14406: Result iterator should not keep reference
+        # for finished futures.
+        for result_object in self.executor.map(_dummy_object_fn, range(10)):
+            self.assertEqual(sys.getrefcount(result_object), 2)
+
+    def test_map_result_order(self):
```

Comment:
Isn't this already tested by `test_map`? Or perhaps you just want to augment `test_map` with a `chunksize`-using test.